src/rosdiscover/observer/ros1.py

```diff
@@ -20,148 +19,48 @@
-                                                   ros, goal_related_topics, result_related_topics):
-                        # Remove the topics from the right collections and replace as an action
-                        goal_related_topics.remove(f"{action}/goal")
-                        goal_related_topics.remove(f"{action}/cancel")
-                        result_related_topics.remove(f"{action}/status")
-                        result_related_topics.remove(f"{action}/feedback")
-                        result_related_topics.remove(f"{action}/result")
-                        actions.append(_Action(action, fmt))
-        return actions
-
-    @staticmethod
-    def has_all_action_topics(action: str,
-                              fmt: str,
-                              ros: ROS1,
-                              goal_related_topics: Collection[str],
-                              result_related_topics: Collection[str]) -> bool:
-        """Check that the non-goal related topics are in the right collections."""
-        return _Node.has_cancel(action, ros, goal_related_topics) and \
-            _Node.has_status(action, ros, result_related_topics) and \
-            _Node.has_feedback(action, fmt, ros, result_related_topics) and \
-            _Node.has_result(action, fmt, ros, result_related_topics)
-
-    @staticmethod
-    def has_cancel(action: str, ros: ROS1, topics: Collection[str]) -> bool:
-        cancel = f"{action}/cancel"
-        try:
-            return cancel in topics and ros.topic_to_type[cancel] == "actionlib_msgs/GoalID"
-        except KeyError:
-            logger.error(f"Topic {cancel} does not have a type.")
-            return False
-
-    @staticmethod
-    def has_status(action: str, ros: ROS1, topics: Collection[str]) -> bool:
-        status = f"{action}/status"
-        try:
-            return status in topics and ros.topic_to_type[
-                status] == "actionlib_msgs/GoalStatusArray"
-        except KeyError:
-            logger.error(f"Topic {status} does not have a type.")
-            return False
-
-    @staticmethod
-    def has_feedback(action: str, fmt: str, ros: ROS1, topics: Collection[str]) -> bool:
-        feedback = f"{action}/feedback"
-        try:
-            return feedback in topics and ros.topic_to_type[feedback] == f"{fmt}Feedback"
-        except KeyError:
-            logger.error(f"Topic {feedback} does not have a type.")
-            return False
-
-    @staticmethod
-    def has_result(action: str, fmt: str, ros: ROS1, topics: Collection[str]) -> bool:
-        result = f"{action}/result"
-        try:
-            return result in topics and ros.topic_to_type[result] == f"{fmt}Result"
-        except KeyError:
-            logger.error(f"Topic {result} does not have a type.")
-            return False
-
 
 class ROS1Observer(Observer):
 
-    def __init__(self, app: AppInstance, config: 'Config'):
+    def __init__(self, app: AppInstance, config: 'Config') -> None:
```

Comment:
There's no need to define this constructor.

Reply:
Done